tsdb/engine/tsm1/DESIGN.md

```diff
@@ -395,25 +395,25 @@ Compactions are IO intensive in that they may need to read multiple, large TSM f
 
 ## Concurrency
 
-The main concern with concurrency is that reads and writes should not block each other.  Writes add entries to the Cache and append entries to the WAL.  During queries, the contention points will be the Cache and existing TSM files.  Since the Cache and TSM file data is only access through the engine by the cursors, several strategies can be used to improve concurrency.
+The main concern with concurrency is that reads and writes should not block each other.  Writes add entries to the Cache and append entries to the WAL.  During queries, the contention points will be the Cache and existing TSM files.  Since the Cache and TSM file data is only accessed through the engine by the cursors, several strategies can be used to improve concurrency.
 
 1. Cache series data can be returned to cursors as a copy.  Since cache entries are evicted on writes, cursors iteration and writes to the same series could block each other.  Iterating over copies of the values can relieve some of this contention.
 2. TSM data values returned by the engine are new references to Values and not access to the actual TSM files.  This means that the `Engine`, through the `FileStore` can limit contention.
-3. Compactions are the only place where new TSM files are added and removed.  Since this is a serial, continously running process, file contention is minimized.
+3. Compactions are the only place where new TSM files are added and removed.  Since this is a serial, continuously running process, file contention is minimized.
 
 ## Robustness
 
 The two robustness concerns considered by this design are writes filling the cache and crash recovery.
 
-Writes filling up cache faster than the WAL segments can be processed result in the oldest entries being evicted from the cache.  This is the normal operation for the cache.  Old entries are always evicited to make room for new entries.  In the case where WAL segements are slow to be processed, writes are not blocked or errored so timeouts should not occur due to IO issues.  A side effect of this is that queries for recent data will always be served from memory.  The size of the in-memory cache can also be tuned so that if IO does because a bottleneck the window of time for queries with recent data can be tuned.
+Writes filling up cache faster than the WAL segments can be processed result in the oldest entries being evicted from the cache.  This is the normal operation for the cache.  Old entries are always evicted to make room for new entries.  In the case where WAL segments are slow to be processed, writes are not blocked or errored so timeouts should not occur due to IO issues.  A side effect of this is that queries for recent data will always be served from memory.  The size of the in-memory cache can also be tuned so that if IO does cause a bottleneck the window of time for queries with recent data can be tuned.
```

Comment:
Wondering if this paragraph is still(/yet to be) true? The current implementation doesn't seem to have an eviction mechanism, other than discarding snapshots once the oldest WAL segments have been compacted. That being the case, it would seem that writes could fill up the hot cache before the compaction associated with the last cache snapshot is complete, resulting in, I think a write error.